packages/react-form-state/src/FormState.ts

```diff
@@ -191,10 +192,18 @@ export default class FormState<
 
     this.setState({submitting: true});
 
+    if (disableOnSubmit) {
+      this.disableFields();
+    }
+
     if (validateOnSubmit) {
       await this.validateForm();
 
       if (this.hasClientErrors) {
+        if (disableOnSubmit) {
```

Comment:
the place where `disableFields` / `enableFields` are called seems to mirror where `submitting` state is changed.
maybe those method can be called in `componentDidUpdate` instead?